docs/readmes/bazel/bazel.md

```diff
@@ -0,0 +1,510 @@
+
+- Protos `BUILD.bazel` files: The global package visibility is set to public as compiled protos libraries need to be accessed from a lot of packages. The visibility of native proto libraries is set to private or is restricted to package visibility for other protos packages.
+- Service `BUILD.bazel` files: The global package visibility is set to private (i.e. default). The binaries should explicitly be set to private for readability reasons, even if this means redundancy. Libraries should explicitly be set to private or package visibility (e.g. for testing) when needed. The public visibility should be used only if necessary.
+- Utility Packages: In the `orc8r/configuration` and `orc8r/common` `BUILD.bazel` files the global package visibility is set to public.
+
+### Target command line syntax
+
+A complete description of the [syntax for specifying targets](https://bazel.build/run/build) can be found in the official bazel documentation. Here we provide a list of some useful examples:
+
+- To build all targets in the bazel workspace, excluding the ones tagged as manual, use `bazel build ...` or `bazel build //...`.
+- To build all targets inside of a folder, e.g. `lte/gateway`, use `bazel build //lte/gateway/...`.
+- Build one target `bazel build //lte/gateway/python/magma/mobilityd:mobilityd`, here the `//` indicate the start of the path to the target and the `:` indicated the start of the target name.
+    - The same target can be build with `bazel build //lte/gateway/python/magma/mobilityd` as this expands to the target with the same name as the folder in the absence of `...` and `:`.
+- To specify multiple targets or multiple folders simply add them to the command line, e.g. `bazel build //orc8r/gateway/... //lte/gateway/python/magma/mobilityd:mobilityd`, this example will build the mobilityd service target, as well as all targets inside the `orc8r/gateway` folder.
+
+A complete list of [command line arguments](https://bazel.build/reference/command-line-reference) can be found in the official bazel documentation.
+
+### Configurations
+
+Targets can be build with different configurations. Configurations are specified on the command line by adding the command line flag `--config=<config_name>`.  Definitions of configurations can be found in the `.bazelrc` file. A configuration is specified by a bazel verb followed by a colon, the name of the configuration and the content, e.g. `build:example --announce_rc --color=yes`.
+
+ The most important configurations for Magma are:
+
+- Plain, i.e. no special configuration, this is the development version of the targets.
+- `production`, which is the production version of the targets used for releases.
+- `asan`, a configuration with the address sanitizer enabled, used for finding memory corruption bugs.
+- `lsan`, a configuration with the leak sanitizer enabled, used for finding run-time memory leaks.
+
+### Dependencies
```

Comment:
+ short paragraphs?
* no cyclic dependencies allowed - mme currently does not allow finer granularity because of that
* design goal: keep targets as dependencies small (better caching) but keep sources in one target that belong together (better maintainability)

Reply:
Paragraphs added